packages/styling/src/interfaces/ISemanticTextColors.ts

```diff
@@ -158,4 +147,14 @@ export interface ISemanticTextColors {
   /** @deprecated
    * This slot was incorrectly named. Use listText instead. */
   listTextColor: string;
+
+  /** @deprecated
```

Comment:
what replaced these if these are deprecated?

Reply:
Maybe you can help me here...

Before my changes the only thing `warningText` was used for was the warning icon. I suspect that it one point it was also styling the text.  Before my change the text for all messageBars was `neutralPrimary`, however I thought `bodyText` would be a better fit. 

Now I'm second guessing that decision, and wondering if this should be another new slot instead.  Since all the MessageBar backgrounds are "invarient" colors, the text color should probably also be "invarient" as well.

What would you think of adding `messageText?`

Reply:
@betrue-final-final maybe this is something that you can help clarify for me.  Which elements of the MessageBar should vary with the current theme, if any?  Should any of the colors be derived from the palette?

Reply:
@phkuo Ok, I've added a few more new slots for this.  Please let me know what you think.

Reply:
Yeah, you shouldn't put neutralPrimary or bodyText on top of message bar background colors. Both of those colors (which should be the same) are designed to go on bodyBackground, not arbitrary backgrounds. Foreground and background sets should "match".

I would prefer we leave warningText in place as the default color for text on message bar backgrounds (instead of introducing a new messageText that does the same thing). Part of the odd name is that we are trying to keep the names agnostic of message bar itself, so it can be used elsewhere, like in forms to highlight issues, or in lists to highlight entire rows.

Reply:
I'd like to understand more.  

Are you saying that I should use `warningText` as the text color for all message bars?  

I was avoiding that, since I thought it would potentially confuse consumers.  The challenge is that in the designs  each message type has its own icon color, but they all have the same text color. So if we have `errorIcon, warningIcon, severeWarningIcon etc`, if you see `errorText, warningText`, you would also expect there to be a `severeWarningText`.  I think I would be a little surprised if the text color inside a "success message" was `warningText`.  That's why I was looking for a more general name to capture all of them.  I chose `message` which I think is still agnostic from the component `messageBar`.

Reply:
I'd prefer we stick with the existing `warningText` since it already exists so theoretically it is already in use by customers. We're trying to NOT add semantic slots when we can.

I agree with your reasoning though. There are a few things I'd love to rename if given a second chance, but it's too late now :(
The best we can do is have thorough documentation and move on.

Reply:
> There are a few things I'd love to rename if given a second chance, but it's too late now :(
> The best we can do is have thorough documentation and move on.

Really??  I understand the desire to not have an explosion of semantic colors, but feeling like the semantic colors can never change seems a bit unhealthy.  Is there someone we can loop in for another opinion on this?

Reply:
Well the purpose of semantic slots is that they have meaning. If they constantly cycle in and out then there's not much point to using them if they're just going to go away the next version. If you're making a button, you need to be sure that `buttonBackground` is going to stick around and not turn into `fluentButtonBackground` while `buttonBackground` gets retired and doesn't get updated anymore.
You can ping me on Teams if you want to discuss this in detail.

Reply:
I understand the desire for stability, however there must be some middle ground between "constantly cycle" and "it's too late now".  That's why there is a deprecation protocol, right?

Are there any guidelines written up about how to manage changes to the theme's semantic color slots?

> the purpose of semantic slots is that they have meaning
Totally agreed, and I'm concerned about moving forward with a slot name thats meaning is inaccurate. Not just by a little, but by a lot.  Instead of `warningText` we could also pick `successText` and from my perspective it would be equally inaccurate.  That said, I acknowledge that different people will pick up different bits of meaning from different words, there's room for interpretation, and that's why I thought it would be helpful to get another opinion in here.

@dzearing @micahgodbolt what do you all think about this?

Reply:
Okay, fine, you've convinced me :)
There are no written guidelines beyond the promise of backwards-compatibility.
My only requirement then is that warningText is set to the value of messageText in the deprecated colors fixup code.

Reply:
> My only requirement then is that warningText is set to the value of messageText in the deprecated colors fixup code.

I can do this, but I'd like to understand the reasoning first, so that I can make sure I'm achieving the intended result. 

`warningText` is not being consumed within the FluentUI codebase, so the way I understand it, our goal would be to minimize disruption for those consumers that are reading the value for their own purposes.  I imagine their expectations are:

1. If I provide a `warningText` color to `createTheme`, the resulting theme should use the color provided. 
2. If I do not provide a value, the value will be the same as before this change. 

With the current code, both of these are true.  With your requirement, I think we could surprise users on both points.

What am I missing?

Reply:
Yes, that is the intention.
1) It still will. Customized semantic slots (and customized _any_ slots) get merged in after _makeSemantic(), on TOP of the result of that call. Therefore any customizations overwrite the default assignments. (theme.ts:L126)
2) But if you provide a value to messageText, you'd want to change the message bar text for consumers of warningText that are trying to achieve the same semantic scenario. When we deprecate, that doesn't mean we want to lock in its default value forever. We still want to support it as much as reasonably possible.

Reply:
Thanks for pointing that out, I missed that detail. I've pushed up the change.

Reply:
Thanks for all your work!